src/main/java/com/chess/backend/restController/service/NewChessGameService.java

```diff
@@ -27,12 +33,13 @@ public NewChessGameService(ChessGameService gameService, GameRepository gameRepo
     /**
      * This method tells the {@link ChessGameService} to create a new game and gets the id of the new game.
      *
-     * @param players the object that is generated via the API-Call of the {@link NewGameController}.
+     * @param newGameObject the object that is generated via the API-Call of the {@link NewGameController}.
      * @return the id of the new created game.
      */
     @Override
-    public int getNewGameID(INewPlayersObject players) {
-        ChessGame game = this.gameService.createNewGame(players.getAllPlayerNames());
+    public int getNewGameID(NewGameObject newGameObject) {
+        List<String> players = Arrays.stream(newGameObject.getPlayers()).map(NewPlayerObject::getPlayerName).collect(Collectors.toList());;
+        ChessGame game = this.gameService.createNewGame((String[]) players.toArray());
```

Comment:
Please fix the casting problem sonarcloud mentioned

Reply:
okidoki